### ISSUE TITLE
To long file name created in DataRequest' "data_folder" 

### ISSUE THREAD

**Issue by PeterFogh** [sentinel-hub/sentinelhub-py#86]
Hi, I have a problem of using the cache functionality. as I get at OSError.

I assume the error is due to the cache file name length excited the 255 character limit in Linux (I use EXT4).

Have you tested such an edge case?
An alternative caching method could be to create a folder structure in the user-defined `data_folder`.
For example for the file: `wms_S2-L1C_EPSG32632_553320.0_6154020.0_553490.0_615_4150.0_2019-04-10T10-41-28_17X13_Downsampling_BILINEAR_EvalScript_return[B01,B02,B03,B04,B05,B08,B09,B10,B11,B12,B8A]_ShowLogo_False_Transparent_True_Upsampling_BILINEAR_tiff_depth=32f.tiff`, (NB: I know it is only 242 character 😅) it could be.
```
-> wms
    |->S2-L1C
        |->EPSG32632
            |->553320.0_6154020.0_553490.0_615_4150.0
                |->2019-04-10T10-41-28_17X13_Downsampling_BILINEAR_EvalScript_return[B01,B02,B03,B04,B05,B08,B09,B10,B11,B12,B8A]_ShowLogo_False_Transparent_True_Upsampling_BILINEAR_tiff_depth=32f.tiff
```

**Comment by AleksMat** [comment]
In version 3.0 we have changed the format of cache filenames. Before we wrote all parameters into the filename. Now we first take all parameters, hash them together into a much shorter string and then store the data in the following structure:
```
-> hashed_string  (a folder)
    | -> request.json
    | -> response.tiff
```
On one hand, we lose readability of the saved data but on the other hand we now also save all request parameters. Therefore, one can simply check `request.json` file to see what the downloaded data refers to.

This has been merged to develop with PR https://github.com/sentinel-hub/sentinelhub-py/pull/106 and will be released today in version `3.0`.

This solves the issue. We'll see in the future how well it will work and if it will still need some improvements.

**Comment by PeterFogh** [comment]
@AleksMat, that sounds like a optimal solution. Nice work😁👍
